### PR TITLE
[IMPROVE] Add pagination to search messages And Fix the search for encrypted room

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

Pagination on SearchMessageView

## Issue(s)	

There isn't pagination when search

## How to test or reproduce

For normal rooms:
- Create a room
- write a lot, if you wanna test with `a`, repeat more than 50 times for the pagination work
- To search you will need to write `/a/` in input to search for `a` in any position of the sentence, this will catch the "a" and the "cat" or the "123a1"

For encrypted rooms:
- Need to activate your E2E encryption
- Then, activate encrypt to the room
- and do the same as before: 
- write a lot, if you wanna test with `a`, repeat more than 50 times for the pagination work
- To search you will need to write `/a/` in input to search for `a` in any position of the sentence, this will catch the "a" and the "cat" or the "123a1"

## Screenshots

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</thead>
</tr>
<tbody>
<tr>
<td>

</td>
<td>

</td>
</tr>
</tbody>
</table>

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/SearchMessagesView/index.js [hunk 1]
```diff
@@ -168,6 +177,16 @@ class SearchMessagesView extends React.Component {
 		}
 	}
 
+	onEndReached = () => {
```
Comment (diegolmello) [inline app/views/SearchMessagesView/index.js:180]: We should be rerunning the request here.

File: app/views/SearchMessagesView/index.js [hunk 2]
```diff
@@ -25,6 +25,11 @@ import database from '../../lib/database';
 import { sanitizeLikeString } from '../../lib/database/utils';
 import getThreadName from '../../lib/methods/getThreadName';
 import getRoomInfo from '../../lib/methods/getRoomInfo';
+import { isTablet } from '../../utils/deviceInfo';
+
+const INITIAL_NUM_TO_RENDER = isTablet ? 20 : 12;
```
Comment (diegolmello) [inline app/views/SearchMessagesView/index.js:30]: Not needed

File: app/views/SearchMessagesView/index.js [hunk 3]
```diff
@@ -53,8 +58,10 @@ class SearchMessagesView extends React.Component {
 		this.state = {
 			loading: false,
 			messages: [],
+			messagesFromQuery: [],
```
Comment (diegolmello) [inline app/views/SearchMessagesView/index.js:61]: We might not need this

File: app/views/SearchMessagesView/index.js [hunk 4]
```diff
@@ -110,12 +117,14 @@ class SearchMessagesView extends React.Component {
 	}
 
 	search = debounce(async(searchText) => {
+		this.count = QUERY_SIZE;
```
Comment (diegolmello) [inline app/views/SearchMessagesView/index.js:120]: Make sure to reset this when search text changes

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/views/SearchMessagesView/index.js [hunk 5]
```diff
@@ -168,6 +173,27 @@ class SearchMessagesView extends React.Component {
 		}
 	}
 
+	onEndReached = async() => {
```
Comment (diegolmello) [inline app/views/SearchMessagesView/index.js:176]: Most of this function is duplicated from `this.search`.
Can we reuse it instead?

File: app/views/SearchMessagesView/index.js [hunk 6]
```diff
@@ -26,6 +26,9 @@ import { sanitizeLikeString } from '../../lib/database/utils';
 import getThreadName from '../../lib/methods/getThreadName';
 import getRoomInfo from '../../lib/methods/getRoomInfo';
 
+const QUERY_SIZE = 50;
+
+
```
Comment (diegolmello) [inline app/views/SearchMessagesView/index.js:31]: Unnecessary empty space

**Review by diegolmello — CHANGES_REQUESTED** [review]
Can you add a test plan for both normal and encrypted rooms?

**Review by AlexAlexandre — APPROVED** [review]
LGTM

**Review by AlexAlexandre — APPROVED** [review]
LGTM

**Review by AlexAlexandre — APPROVED** [review]
LGTM :)

**Pull request merged**